juju/juju
# Add model type to client models.xml; various model apis also return model type

## Description of change

A largely mechanical PR, we add model-type to the client models.xml file. This is so that various CLI commands can find out what type of model they are operating on.
Various model API calls need then to include model type in their result. This is included in show/list model output.
We also rename add-caas to add-k8s

## QA steps

bootstrap
check list/show model for controller and default models
create a caas model
check list/show model
verify content of models.xml file
